sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryStorageStreamBundleSource.java

```diff
@@ -160,8 +161,9 @@ public long getEstimatedSizeBytes(PipelineOptions options) {
   @Override
   public List<? extends OffsetBasedSource<T>> split(
       long desiredBundleSizeBytes, PipelineOptions options) {
-    // A stream source can't be split without reading from it due to server-side liquid sharding.
-    // TODO: Implement dynamic work rebalancing.
+    // This method is only called for initial splits. Since this class will always be a child source
+    // of BigQueryStorageSourceBase, all splits here will be handled by `splitAtFraction()`. As a
+    // result, this is a no-op.
```

Comment:
FYI because it was mentioned previously as `TODO: Implement dynamic work rebalancing`, splitAtFraction() _is_ dynamic work rebalancing